### PR TITLE
RaviTella cosmos diagnostics activityid missing fix

### PR DESCRIPTION
**Pull request by RaviTella**
Included "activityId" in diagnostics logs. Currently it only shows up in query diagnostics. 

Before:
	"storeResult": {
			"storePhysicalAddress": "rntbd://cdb-ms-prod-westus1-fd20.documents.azure.com:14426/apps/1ef6f60b-b95b- 40b8-8a46-543f7f6322d8/services/b5c02c75-0d2b-4b97-80f6-6e47bb494c46/partitions/75a6954e-1dd6-42ba-9706-f0cb9d585252/replicas/132765133258229411p/",
			"lsn": 1910,
			"globalCommittedLsn": 1910,
			"partitionKeyRangeId": "1",
			"isValid": true,
			"statusCode": 200,
			"subStatusCode": 0,
			"isGone": false,
			"isNotFound": false,
			"isInvalidPartition": false,
			"isThroughputControlRequestRateTooLarge": false,
			"requestCharge": 2.71,
			"itemLSN": -1,

After fix:
	"responseStatisticsList": [{
		"storeResult": {
			"storePhysicalAddress": "rntbd://cdb-ms-prod-westus1-fd20.documents.azure.com:14111/apps/1ef6f60b-b95b-40b8-8a46-543f7f6322d8/services/b5c02c75-0d2b-4b97-80f6-6e47bb494c46/partitions/75a6954e-1dd6-42ba-9706-f0cb9d585252/replicas/132777309365865748s/",
			"lsn": 1910,
			"globalCommittedLsn": 0,
			"partitionKeyRangeId": "1",
			"isValid": true,
			"statusCode": 200,
			"subStatusCode": 0,
			"isGone": false,
			"isNotFound": false,
			"isInvalidPartition": false,
			"isThroughputControlRequestRateTooLarge": false,
			"requestCharge": 2.71,
			**"activityId": "c207fb31-2adb-11ec-b1a0-bd8070754eed",**
			"itemLSN": -1,

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
QQ: How to ensure that required details are always present? 

One option is to code it as assumption in CTL and failure is a bug.
Its a POJO verification so perf impact also negligible. 

Thoughts?

/cc: @moderakh , @simplynaveen20 , @FabianMeiswinkel , @ealsur

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java [hunk 1]
```diff
@@ -394,13 +397,15 @@ private void validateDirectModeQueryDiagnostics(String diagnostics) {
         assertThat(diagnostics).contains("\"gatewayStatistics\":null");
         assertThat(diagnostics).contains("addressResolutionStatistics");
         assertThat(diagnostics).contains("\"userAgent\":\"" + Utils.getUserAgent() + "\"");
+        assertThat(diagnostics).contains("\"activityId\":\"");
```
[the remark this answers is not in the document]
Comment (RaviTella) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java]: Updated the test to assert for the existence of non empty value for activityId

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java [hunk 2]
```diff
@@ -238,7 +241,7 @@ public void directDiagnostics() {
             assertThat(diagnostics).contains("\"metaDataName\":\"PARTITION_KEY_RANGE_LOOK_UP\"");
             assertThat(diagnostics).contains("\"metaDataName\":\"SERVER_ADDRESS_LOOKUP\"");
             assertThat(diagnostics).contains("\"serializationType\":\"PARTITION_KEY_FETCH_SERIALIZATION\"");
-            assertThat(diagnostics).contains("\"userAgent\":\"" + Utils.getUserAgent() + "\"");
```
[the remark this answers is not in the document]
Comment (RaviTella) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java:241]: Restored the assertion for UserAgent.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

**Review by kushagraThapar — APPROVED** [review]

**Comment by RaviTella** [comment]
This fixes the "missing activityId in request diagnostics" bug.

**Pull request merged**